### ISSUE TITLE
Multiple issues with v.3

### ISSUE THREAD

**Issue by nettunodev** [barbushin/php-imap#321]
Hello, I was using the v. 2.0.8 and I upgraded to v. 3.0.6. But then I found out there was an issue with certified emails, sometimes some attachments had zero-byte size.

Then I tried to upgrade to v. 3.0.15 and I encountered the following issues:
- on php 7.1.8 and iconv v. 1.14 the function `getMailsInfo` fail silently, even with all errors/warnings/notice enabled. I replaced the line in `convertStringEncoding` this way

```
//$convertedString = function_exists('iconv') ? @iconv($fromEncoding, $toEncoding . '//IGNORE', $string) : null;
$convertedString = false;
```

However it does work on php 7.0.3 and iconv v. 2.23. Also, this didn't happen with v. 3.0.6.
- since v. 3.0.11 the attachments are not saved on disk anymore. I didn't change my code and I tried all the php-imap releases from 3.0.15 downgrading until v. 3.0.10 and finally the attachments were being saved on disk again. 

Just to be safe I now downgraded to v. 2.0.8.

**Comment by Sebi94nbg** [comment]
The issue regarding `iconv()` might be the same issue as described in #320.

Regarding the attachments: In v3.0.11 were the ability to skip attachments added. See CHANGELOG: https://github.com/barbushin/php-imap/releases/tag/3.0.11

From the [wiki](https://github.com/barbushin/php-imap/wiki/Getting-Started#disable-processing-of-attachments-to-boost-performance):
> By default, all fetched emails will be parsed to extract the attachments.

Attachments will be only saved to the disk, when you specify the `$attachmentsDir` in either the constructor...
```php
$mailbox = new PhpImap\Mailbox(
	'{imap.gmail.com:993/imap/ssl}INBOX', // IMAP server and mailbox folder
	'some@gmail.com', // Username for the before configured mailbox
	'*********', // Password for the before configured username
	__DIR__, // Directory, where attachments will be saved (optional)
	'US-ASCII' // Server encoding (optional)
);
```
...or using the method:
```php
// Change attachments directory
// Useful, when you did not set it at the beginning or
// when you need a different folder for eg. each email sender
$mailbox->setAttachmentsDir('/var/www/example.com/ticket-system/imap/attachments');
```

Is this path configured in your case? You can check this by getting the current set value:
```php
echo "Save attachments to: " . $mailbox->getAttachmentsDir() . "\n";
```

How does your code look like?

**Comment by Sebi94nbg** [comment]
Just tried it by my own: Path is set correctly, but `iconv()` fails:
> PHP Notice:  iconv(): Wrong charset, conversion from \`default' to \`UTF-8//IGNORE' is not allowed in /home/sebastian/git/php-imap/src/PhpImap/Mailbox.php on line 1013

I'll check and fix this issue.

**Comment by Sebi94nbg** [comment]
@nettunodev can you please test my two changes, if they fix all your problems?

Use the current code of the `develop` branch:
`composer require php-imap/php-imap:dev-develop`

When it's working fine, I'll merge it into master and create a new release.

**Comment by tomnuss** [comment]
@Sebi94nbg FYI - just started testing with attachments today, and wasn't able to save attachments with the latest version. Just tested the develop branch, and I'm now able to save attachments. Thanks.

**Comment by Sebi94nbg** [comment]
Great. Thanks for verifying the attachment part, @tomnuss!

@nettunodev only needs to verify the issue with `iconv()` and then we can close this issue. :)

**Comment by nettunodev** [comment]
Hello, thanks for the prompt reply! I can confirm that everything seems to work now. Also, the bug that i encountered in v. 3.0.6 seems to be fixed, but I have to further test this. I can always open another issue if I find other bugs. Thank you!

**Comment by Sebi94nbg** [comment]
Thanks for your feedback! I've released [v3.0.16](https://github.com/barbushin/php-imap/releases/tag/3.0.16) with those fixes. :)
